leshan-core/src/main/java/org/eclipse/leshan/core/attr/AttributeSet.java

```diff
@@ -0,0 +1,77 @@
+package org.eclipse.leshan.core.attr;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.leshan.core.request.exception.InvalidRequestException;
+
+/**
+ * A collection of {@link Attribute} instances that are handled as a unit regarding validation etc.
+ */
+public class AttributeSet {
+    
+    private static final String PARAM_SEP = "&";
+    private static final String CANCEL = "cancel";
```

Comment:
I did search so much but I did not find any reference to cancel attribute. Could you point where this is defined in the spec ?

Reply:
Not me either :) I saw that this construct existed in ObserveSpec. I'll remove it for now.

Reply:
Probably out of date you can remove it.